runtime-sdk/modules/evm/src/lib.rs

```diff
@@ -429,18 +430,26 @@ impl<Cfg: Config> Module<Cfg> {
         H160::from_slice(&out[32 - 20..])
     }
 
-    fn derive_caller_from_tx_auth_info(ai: &AuthInfo) -> Result<H160, Error> {
-        match &ai.signer_info[0].address_spec {
-            AddressSpec::Signature(PublicKey::Secp256k1(pk)) => {
-                Ok(Self::derive_caller_from_bytes(pk.as_bytes()))
-            }
-            AddressSpec::Signature(PublicKey::Ed25519(_)) => Ok(Self::derive_caller_from_bytes(
+    fn derive_caller_from_public_key(pk: &PublicKey) -> Result<H160, Error> {
+        match pk {
+            PublicKey::Secp256k1(pk) => Ok(Self::derive_caller_from_bytes(
+                &pk.uncompressed_untagged_bytes()
+                    .ok_or(Error::InvalidArgument)?,
+            )),
+            PublicKey::Ed25519(_) => Ok(Self::derive_caller_from_bytes(
                 &ai.signer_info[0].address_spec.address().as_ref()[1..],
             )),
             _ => Err(Error::InvalidSignerType),
         }
     }
 
+    fn derive_caller_from_tx_auth_info(ai: &AuthInfo) -> Result<H160, Error> {
+        match &ai.signer_info[0].address_spec {
+            AddressSpec::Signature(pk) => Self::derive_caller_from_public_key(pk),
```

Comment:
For all non-secp256k1 specs the caller should be derived from the address regardless of what scheme it is using.

Reply:
that's a divergence from what this previously did, but we can go ahead and do that as part of this PR

Reply:
added as a commit near the beginning